cli/src/cmd/forge/script/cmd.rs

```diff
@@ -46,137 +48,224 @@ impl ScriptArgs {
 
-            verify.set_chain(&script_config.config, chain.into());
+        if self.debug {
+            return self.run_debugger(&decoder, sources, result, project, highlevel_known_contracts)
+        }
 
-            let broadcasted = self.broadcast || self.resume;
-            let mut deployment_sequence = match ScriptSequence::load(
-                &script_config.config,
-                &self.sig,
-                &target,
-                chain,
-                broadcasted,
-            ) {
-                Ok(seq) => seq,
-                // If the script was simulated, but there was no attempt to broadcast yet,
-                // try to read the script sequence from the `dry-run/` folder
-                Err(_) if broadcasted => {
-                    ScriptSequence::load(&script_config.config, &self.sig, &target, chain, false)?
-                }
-                Err(err) => eyre::bail!(err),
-            };
-
-            receipts::wait_for_pending(provider, &mut deployment_sequence).await?;
-
-            if self.resume {
-                self.send_transactions(&mut deployment_sequence, &fork_url, result.script_wallets)
-                    .await?;
-            }
+        self.maybe_prepare_libraries(
+            &mut script_config,
+            project,
+            &mut highlevel_known_contracts,
+            default_known_contracts,
+            (&mut libraries, predeploy_libraries),
+            (&mut result, &mut decoder),
+        )
+        .await?;
+
+        if self.json {
+            self.show_json(&script_config, &result)?;
+        } else {
+            self.show_traces(&script_config, &decoder, &mut result).await?;
+        }
 
-            if self.verify {
-                // We might have predeployed libraries from the broadcasting, so we need to relink
-                // the contracts with them, since their mapping is not included in the solc cache
-                // files.
-                let BuildOutput { highlevel_known_contracts, .. } = self.link(
+        verify.known_contracts = flatten_contracts(&highlevel_known_contracts, false);
+        self.check_contract_sizes(&result, &highlevel_known_contracts)?;
+
+        self.handle_broadcastable_transactions(
+            result,
+            libraries,
+            &mut decoder,
+            script_config,
+            verify,
+        )
+        .await
+    }
+
+    // In case there are libraries to be deployed, it makes sure that these are added to the list of
+    // broadcastable transactions  with the appropriate sender.
+    async fn maybe_prepare_libraries(
+        &mut self,
+        script_config: &mut ScriptConfig,
+        project: Project,
+        highlevel_known_contracts: &mut ArtifactContracts<ContractBytecodeSome>,
+        default_known_contracts: ArtifactContracts,
+        (libraries, predeploy_libraries): (&mut Libraries, Vec<Bytes>),
+        (result, decoder): (&mut ScriptResult, &mut CallTraceDecoder),
```

Comment:
this is a bit much can we use a helper type here.

since this function replaces the values it can just take ownership and return the modified values